website: Update the contributing.md for GitHub

BobbyRBruce opened the pull request.

BobbyRBruce committed:
website: Update the contributing.md for GitHub

powerjg commented on _pages/contributing.md:
```diff
@@ -275,136 +303,47 @@ Jira Issue: https://gem5.atlassian.net/browse/GEM5-186
-For non-trivial changes, it is not unusual for a change to receive feedback
-from reviewers that they will want incorporated before giving the commit a
-score necessary for it to be merged. This leads to an iterative process.
+Once your pull-request has passed all the CI tests and has been approved by at least one community member, it will be merged a gem5 Project Maintainer will do a [Squash and Merge](https://docs.gitlab.com/ee/user/project/merge_requests/squash_and_merge.html) on the pull-request.
```
Wrong link :)

```suggestion
Once your pull-request has passed all the CI tests and has been approved by at least one community member, it will be merged a gem5 Project Maintainer will do a [Squash and Merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits) on the pull-request.
```

BobbyRBruce committed:
website: Fix incorrect squash-and-merge link

Co-authored-by: Jason Lowe-Power <jason@lowepower.com>

BobbyRBruce committed:
Merge branch 'stable' into update-contributing

BobbyRBruce committed:
website: Move and expand 'Keeping your forked repo..."

This expands how to keep your forked and local repository up-to-date
(i.e., in-synic with the main gem5 repository).

It also moves it to it's own section.

BobbyRBruce committed:
website: Fix 'stable' -> 'develop' typo

BobbyRBruce committed:
website: 'recommend' -> 'strongly recommend' for local branches

BobbyRBruce committed:
website: Minor changes to contributing.md
